proposals/0000-initial-extension-categorization.md

```diff
@@ -0,0 +1,224 @@
+| UnboxedTuples              | Stable         | Stable                  |
+| UnicodeSyntax              | Stable         | Stable                  |
+| UnliftedDatatypes          | Stable         | Stable                  |
+| UnliftedFFITypes           | Stable         | Stable                  |
+| UnliftedNewtypes           | Stable         | Stable                  |
+| ViewPatterns               | Stable         | Stable                  |
+
+## 2.2 Extensions Impacting Stablity Categorization
+
+Here we list extensions where the categorization is different based on if the extension is enabled or not. This separation is specifically to make reading the tables of extensions easier. We have further divided this section into `Legacy` and `Experimental` extensions.
+
+### 2.2.1 Legacy Extensions
+
+The following are all extensions that are to be categorized as `Legacy` when enabled, and for the `No` version are categorized as `Stable`. As a reminder, the [categorizations](https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0601-extension-lifecycle-framework.md#21-categories) state that `Legacy` extensions are not recommended for new code but are expected to be supported indefinitely.
+
+The following table contains the language extension, the proposed categorization for the extension being turned on, and the proposed categorization for the extension being turned off.
+
+| Extension                | Categorization | Negative Categorization |
+|--------------------------|----------------|-------------------------|
+| CUSKs                    | Legacy         | Stable                  |
+| DatatypeContexts         | Legacy         | Stable                  |
+| DeepSubsumption          | Legacy         | Stable                  |
+| NondecreasingIndentation | Legacy         | Stable                  |
+| NPlusKPatterns           | Legacy         | Stable                  |
+| StarIsType               | Legacy         | Stable                  |
+
+### 2.2.2 Experimental Extensions
+
+The following are extensions that are to be categorized as `Experimental` when enabled, and for the `No` or negative version are categorized as `Stable`. As a reminder, the [categorizations](https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0601-extension-lifecycle-framework.md#21-categories) state that `Experimental` extensions "are undergoing active development, or have consequences that make it impossible in practice to avoid breaking changes".
+
+The following table contains the language extension, the proposed categorization for the extension being turned on, and the proposed categorization for the extension being turned off.
+
+| Extension              | Categorization | Negative Categorization |
+|------------------------|----------------|-------------------------|
+| ApplicativeDo          | Experimental   | Stable                  |
+| Arrows                 | Experimental   | Stable                  |
```

Comment:
I don't like the presence of `Arrows` and `ApplicativeDo` here, or the word "experimental". These extensions aren't experimental, there is no experiment, they just have a bunch of long-standing bugs that someone may decide to solve at some point in the future

Reply:
I'm objecting simply because of the implications. I like the "preview" term because it doesn't imply an experiment, old "preview" extension just means that it's stale/frozen in its preview form.

Reply:
My stance is that we could spin on the exact terms and never make any progress on other things. As @adamgundry points out there is some compromise on any such terminology. As stated in the PR message, in order to make some progress on the majority, I think it's best to move controversial choices to uncategorized and revisit later.

Reply:
I've moved both of these to uncategorized for the purposes of the this proposal.